lib/src/node/value/calculation.dart

```diff
@@ -0,0 +1,144 @@
+
+import 'dart:js_util';
+
+import 'package:collection/collection.dart';
+import 'package:node_interop/js.dart';
+import 'package:sass/src/node/immutable.dart';
+import 'package:sass/src/node/utils.dart';
+
+import '../../value.dart';
+import '../reflection.dart';
+
+/// Check that [arg] is a valid argument to a calculation function.
+void isCalculationValue(Object arg, {bool checkUnquoted = true}) {
+  if (arg is! SassNumber &&
+      arg is! SassString &&
+      arg is! SassCalculation &&
+      arg is! CalculationOperation &&
+      arg is! CalculationInterpolation) {
+    jsThrow(JsError('Argument must be one of '
+        'SassNumber, SassString, SassCalculation, CalculationOperation, '
+        'CalculationInterpolation'));
+  }
+  if (checkUnquoted && arg is SassString && arg.hasQuotes) {
+    jsThrow(JsError('Argument must be unquoted SassString'));
+  }
+}
+
+/// The JavaScript `SassCalculation` class.
+final JSClass calculationClass = () {
+  calc(Object argument) {
+    isCalculationValue(argument);
+    return SassCalculation.unsimplified('calc', [argument]);
+  }
+
+  min(Object arguments) {
+    var argList = jsToDartList(arguments).cast<Object>();
+    argList.forEach(isCalculationValue);
+    return SassCalculation.unsimplified('min', argList);
+  }
+
+  max(Object arguments) {
+    var argList = jsToDartList(arguments).cast<Object>();
+    argList.forEach(isCalculationValue);
+    return SassCalculation.unsimplified('max', argList);
+  }
+
+  clamp(Object min, [Object? value, Object? max]) {
+    if (value == null && max != null) {
+      jsThrow(JsError('`value` is undefined and `max` is not undefined'));
+    }
+    if ((value == null || max == null) &&
+        !(min is SassString && min.text.startsWith('var(')) &&
+        !(value is SassString && value.text.startsWith('var('))) {
+      jsThrow(JsError(
+          '`value` or `max` is undefined and neither `min` nor `value` is a SassString that begins with "var("'));
+    }
+    [min, value, max].whereNotNull().forEach(isCalculationValue);
+    return SassCalculation.unsimplified(
+        'clamp', [min, value, max].whereNotNull());
+  }
+
+  var jsClass =
+      createJSClass('sass.SassCalculation', (Object self, [Object? _]) {
+    jsThrow(JsError("new sass.SassCalculation() isn't allowed"));
+  });
+
+  // Static methods
+  setProperty(jsClass, 'calc', allowInteropNamed('calc', calc));
+  setProperty(jsClass, 'min', allowInteropNamed('min', min));
+  setProperty(jsClass, 'max', allowInteropNamed('max', max));
+  setProperty(jsClass, 'clamp', allowInteropNamed('clamp', clamp));
+
+  jsClass.defineMethods({
+    'assertCalculation': (SassCalculation self, [String? name]) => self,
+  });
+
+  jsClass.defineGetters({
+    'name': (SassCalculation self) => self.name,
+    'arguments': (SassCalculation self) => self.arguments,
+  });
+
+  getJSClass(SassCalculation.unsimplified('calc', [SassNumber(1)]))
+      .injectSuperclass(jsClass);
```

Comment:
I got this approach from [`color.dart`](https://github.com/oddbird/dart-sass/blob/f5ce0906e7013eace2800f601870cb80193de712/lib/src/node/value/color.dart#L67), is it correct?